app/javascript/components/modals/complete-exercise-modal/PublishSolutionModal.tsx

```diff
@@ -17,18 +18,23 @@ export const PublishSolutionModal = ({
   endpoint: string
   onSuccess: (data: ExerciseCompletion) => void
 }): JSX.Element => {
+  const ariaObject = generateAriaFieldIds('publish-code')
   return (
     <Modal
       cover={true}
       open={open}
+      role="dialog"
+      aria={ariaObject}
```

Comment:
1. Are there modals where we **don't** want these? If so, we could consider moving them upstream. 
2. Consider passing `ariaId` in here, with the value of `publish-code` and move the `generateAriaFieldIds` call into `Modal` itself. That'll save having to have this same `const ariaObject = generateAriaFieldIds(...)` line of code scattered throughout the codebase.